index.js

```diff
@@ -1,245 +1,9 @@
-          return img;
-        });
-
-        imagesWithSizes.sort(heightAscThanNameComparator);
-
-        var sprite = new ShelfPack(1, 1, { autoResize: true });
-        sprite.pack(imagesWithSizes, { inPlace: true });
-
-        if (options.format) {
-            var obj = {};
-            imagesWithSizes.forEach(function(item) {
-                var itemIdsToUpdate = [item.id];
-                if (options.unique) {
-                    var svg = svgPerItemId[item.id];
-                    itemIdsToUpdate = itemIdsPerSvg[svg];
-                }
-                itemIdsToUpdate.forEach(function(itemIdToUpdate) {
-                    obj[itemIdToUpdate] = {
-                        width: item.width,
-                        height: item.height,
-                        x: item.x,
-                        y: item.y,
-                        pixelRatio: options.pixelRatio
-                    };
-                });
-            });
-            return callback(null, obj);
-
-        } else {
-            return callback(null, {
-                width: sprite.w,
-                height: sprite.h,
-                items: imagesWithSizes
-            });
-        }
-
-    });
-}
-
-
-/**
- * Generate a PNG image with positioned icons on a sprite.
- *
- * @param  {ImgLayout}   layout    An {@link ImgLayout} Object used to generate the image
- * @param  {Function}    callback  Accepts two arguments, `err` and `image` data
- */
-function generateImage(layout, callback) {
-    assert(typeof layout === 'object' && typeof callback === 'function');
-    if (!layout.items.length) return callback(null, emptyPNG);
-
-    mapnik.blend(layout.items, {
-        width: layout.width,
-        height: layout.height
-    }, callback);
-}
-
-
-/**
- * Spritezero can generate 2 kinds of layout objects: {@link DataLayout} and {@link ImgLayout}.
- *
- * A `ImgLayout` Object contains the array of image items along with dimensions
- * and a buffer of image data that can be used for generating the output image.
- *
- * @typedef  {Object}    ImgLayout
- * @example
- * {
- *    width: 512,
- *    height: 512,
- *    items: [
- *      {
- *        "height": 12,
- *        "width": 12,
- *        "x": 133,
- *        "y": 282,
- *        "buffer": "..."
- *      }, ... etc ...
- *    ]
- * }
- */
-
-
-/**
- * Spritezero can generate 2 kinds of layout objects: {@link DataLayout} and {@link ImgLayout}.
- *
- * A `DataLayout` Object contains all the metadata about the contents of the sprite.
- * This data can be exported to a JSON sprite manifest file.
- *
- * The keys of the Object are the icon ids.
- * The values of the Object are the structured data about each icon.
- *
- * @typedef  {Object}    DataLayout
- * @example
- * {
- *    "aerialway-12": {
- *      "width": 12,
- *      "height": 12,
- *      "pixelRatio": 1,
- *      "x": 133,
- *      "y": 282
- *    }, ... etc ...
- * }
- */
+exports.validateMetadata = require('./lib/validate-svg-metadata');
```

Comment:
This is fine, but would have been easier to review this PR if you separated this restructure from your new feature!

Reply:
I moved the file, but Git didn't pick it up and shows the combined diff as removed + added, rather than renamed. You can now look at the individual commits which correctly track the rename though.